common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/MqttTransportHandler.java

```diff
@@ -1330,6 +1340,37 @@ public void onToDeviceRpcRequest(UUID sessionId, TransportProtos.ToDeviceRpcRequ
         }
     }
 
+    private void onGetServiceConfigurationRpc( TransportProtos.SessionInfoProto sessionInfo, ChannelHandlerContext ctx, int msgId, TransportProtos.ToServerRpcRequestMsg rpcRequestMsg) {
+        var tenantProfile = context.getTenantProfileCache().get(deviceSessionCtx.getTenantId());
+        DefaultTenantProfileConfiguration profile = tenantProfile.getDefaultProfileConfiguration();
+        Map<String, Object> serviceConfiguration = new HashMap<>();
```

Comment:
Please create a proper object structure instead of the map (POJO) . It will make code more readable and simple. Also I don't like the duplication of the 'RateLimit' and 'gateway/device' in every parameter. Seems redundant.